app/controllers/user_avatars_controller.rb

```diff
@@ -126,7 +126,11 @@ def show_in_site(hostname)
         optimized_path = Discourse.store.path_for(optimized)
         image = optimized_path if File.exist?(optimized_path)
       else
-        return proxy_avatar(Discourse.store.cdn_url(optimized.url), upload.created_at)
+        if GlobalSetting.redirect_avatar_requests
+          return redirect_s3_avatar(Discourse.store.cdn_url(optimized.url))
+        else
+          return proxy_avatar(Discourse.store.cdn_url(optimized.url), upload.created_at)
+        end
```

Comment:
merging ifs (can't make a code suggestion because line 128 wasn't touched 😆)

```rb
     elsif GlobalSetting.redirect_avatar_requests
       return redirect_s3_avatar(Discourse.store.cdn_url(optimized.url))
     else
       return proxy_avatar(Discourse.store.cdn_url(optimized.url), upload.created_at)
     end
```